packages/salesforcedx-vscode-apex/src/codecoverage/colorizer.ts

```diff
@@ -0,0 +1,176 @@
+  if (!fs.existsSync(testRunIdFile)) {
+    throw new Error('No test run information was found for this project.');
+  }
+  return fs.readFileSync(testRunIdFile, 'utf8');
+}
+
+function getCoverageData() {
+  const testRunId = getTestRunId();
+  const testResultFilePath = path.join(
+    apexDirPath,
+    `test-result-${testRunId}.json`
+  );
+
+  if (!fs.existsSync(testResultFilePath)) {
+    throw new Error(
+      `No code coverage information was found for test run ${testRunId}`
+    );
+  }
+  const testResultOutput = fs.readFileSync(testResultFilePath, 'utf8');
+  const codeCoverage = JSON.parse(testResultOutput) as CoverageTestResult;
+  if (codeCoverage.coverage === undefined) {
+    throw new Error(
+      `No code coverage information was found for test run ${testRunId}.`
+    );
+  }
+  return codeCoverage.coverage ? codeCoverage.coverage.coverage : '';
+}
+
+function isApexMetadata(filePath: string): boolean {
+  return filePath.endsWith('.cls') || filePath.endsWith('.trigger');
+}
+
+function getApexMemberName(filePath: string): string {
+  if (isApexMetadata(filePath)) {
+    const filePathWithOutType = filePath.replace(/.cls|.trigger/g, '');
+    const indexOfLastFolder = filePathWithOutType.lastIndexOf('/');
+    return filePathWithOutType.substring(indexOfLastFolder + 1);
+  }
+  return '';
+}
+
+export class CodeCoverage {
+  private statusBar: StatusBarToggle;
+  public coveredLines: Range[];
+  public uncoveredLines: Range[];
+
+  constructor(statusBar: StatusBarToggle) {
+    this.statusBar = statusBar;
+    this.coveredLines = Array<Range>();
+    this.uncoveredLines = Array<Range>();
+
+    window.onDidChangeActiveTextEditor(this.onDidChangeActiveTextEditor, this);
+    this.onDidChangeActiveTextEditor(window.activeTextEditor);
+  }
+
+  public onDidChangeActiveTextEditor(editor?: TextEditor) {
+    if (editor && this.statusBar.isHighlightingEnabled) {
+      this.colorizer(editor);
+    }
+  }
+
+  public toggleCoverage() {
+    if (this.statusBar.isHighlightingEnabled) {
+      this.statusBar.toggle(false);
+      this.coveredLines = [];
+      this.uncoveredLines = [];
+
+      const editor = window.activeTextEditor;
+      if (editor) {
+        editor.setDecorations(coveredLinesDecorationType, this.coveredLines);
+        editor.setDecorations(
+          uncoveredLinesDecorationType,
+          this.uncoveredLines
+        );
+      }
+    } else {
+      this.colorizer(window.activeTextEditor);
+      this.statusBar.toggle(true);
+    }
+  }
+
+  public colorizer(editor?: TextEditor) {
+    try {
+      if (editor && isApexMetadata(editor.document.uri.fsPath)) {
+        const codeCovArray = getCoverageData() as CoverageItem[];
+        const codeCovItem = codeCovArray.find(
+          covItem =>
+            covItem.name === getApexMemberName(editor.document.uri.fsPath)
+        );
+
+        if (!codeCovItem) {
+          throw new Error(
+            'No code coverage information was found for the current file.'
```

Comment:
We'll show this when the test results do have code coverage information but not for the apex file currently in focus.